### ISSUE TITLE
scd30.getMeasurementInterval() returns wrong value

### ISSUE THREAD

**Issue by robinschneider** [adafruit/Adafruit_SCD30#12]
I used the sensor_tuning.ino example and when I set the measurement interval to 5 or 10 the return values are wrong.

For example if I set the interval to 5 seconds the output of
```
 Serial.print("Measurement interval: ");
 Serial.print(scd30.getMeasurementInterval());
 Serial.println(" seconds");
```
is `Measurement interval: 17920 seconds`

The measurement interval will be 5 seconds, but the return value is wrong.
Only worked when setting it back to 2 seconds.

**Comment by robinschneider** [comment]
I used an interval of 5 seconds, 360 meters altitude offset, 310 (3.1°C) temperature offset
```
Adafruit SCD30 Sensor adjustment test!
SCD30 Found!
Measurement interval: 17920 seconds
Ambient pressure offset: 0 mBar
Altitude offset: 360 meters
Temperature offset: 215.07 degrees C
Forced Recalibration reference: 400 ppm
Self calibration disabled
```

**Comment by caternuson** [comment]
What main board are you using to connect the SCD 30 to? Just did a quick check using that same example and can't recreate this behavior. It returns the expected 5 seconds.
![Screenshot from 2021-04-21 11-17-45](https://user-images.githubusercontent.com/8755041/115601946-3c455100-a293-11eb-9eb6-b93cb36e1a7b.png)

**EDIT** above test was with a QT Py (samd21) and connected via STEMMA cable.

**Comment by robinschneider** [comment]
I used an ESP32

**Comment by caternuson** [comment]
Ran the sketch on a Feather ESP32 and still can not recreate.
![Screenshot from 2021-04-21 11-56-29](https://user-images.githubusercontent.com/8755041/115606549-b0362800-a298-11eb-8786-5e35f0a75165.png)

What exact ESP32 hardware are you using and how is the SCD30 connected to it?

**Comment by robinschneider** [comment]
I used a DOIT Esp32 DevKit v1

**Comment by caternuson** [comment]
Do you have another board available you could try using to see if the problem shows up again? Since we can't recreate it, seems like something specific to your setup.

**Comment by robinschneider** [comment]
I have two of them, both new and both have the same issue.
Also I tried it with plattformio and arduino ide.
Another problem I have is when setting a measurement interval, altitude or temperature offset, most of the time it fails.
Sometimes I restart the ESP32 and then the values are different than before.

**Comment by caternuson** [comment]
` Sometimes I restart the ESP32 and then the values are different than before.`
That could be this #13

**Comment by robinschneider** [comment]
So I have an Arduino Uno and it works without any of the issues mentioned above.
So at least my sensor seems to be fine, but I still wonder what happens.
I wanted to send the measurements over MQTT.

**Comment by caternuson** [comment]
But at this point you aren't doing anything with MQTT, right? Just running `sensor_tuning.ino` and seeing that behavior where it won't return what was set?

**Comment by robinschneider** [comment]
Yes, I just run the `sensor_tuning.ino` and set the measurement interval to 5 seconds, altitude offset to 360 and temperature to 175.

**Comment by caternuson** [comment]
How is the SCD30 connected to the ESP32? What pins? What kind of wires? etc.

**Comment by robinschneider** [comment]
I used I2C to connect them. For power, I used the 3V3 and GND pin of the ESP32, SCL D22 and SDA D21

**Comment by caternuson** [comment]
That seems fine. If you have a second ESP32, or a second 3V level board, maybe try that and see if the issue repeats. It doesn't seem to be the SCD30, since it works OK on the UNO. It doesn't appear to be a library or ESP32 issue, since we couldn't repeat it.

**Comment by robinschneider** [comment]
So I tested some more ESP32 boards and I have the same issue everywhere.
I tried:
- 2 DOIT Esp32 DevKit v1
- 2 FireBeetle ESP32-E
- 1 Adafruit HUZZAH32
Most of the time I dont't even get the values because it failes before.
```
Adafruit SCD30 Sensor adjustment test!
SCD30 Found!
Failed to set measurement interval
```
Not only the measurement interval often fails, also the temperature and altitude offset.  
I tried it with 3.3V and 5V.  
```
// Demo of the available settings for the SCD30
#include <Adafruit_SCD30.h>

Adafruit_SCD30  scd30;

void setup(void) {
  Serial.begin(115200);
  while (!Serial) delay(10);     // will pause Zero, Leonardo, etc until serial console opens

  Serial.println("Adafruit SCD30 Sensor adjustment test!");

  // Try to initialize!
  if (!scd30.begin()) {
    Serial.println("Failed to find SCD30 chip");
    while (1) { delay(10); }
  }
  Serial.println("SCD30 Found!");

  /***
   * The code below will report the current settings for each of the
   * settings that can be changed. To see how they work, uncomment the setting
   * code above a status message and adjust the value
   *
   * **Note:** Since Automatic self calibration and forcing recalibration with
   * a reference value overwrite each other, you should only set one or the other
  ***/


  /*** Adjust the rate at which measurements are taken, from 2-1800 seconds */
  if (!scd30.setMeasurementInterval(10)) {
    Serial.println("Failed to set measurement interval");
    while(1){ delay(10);}
  }
  Serial.print("Measurement interval: ");
  Serial.print(scd30.getMeasurementInterval());
  Serial.println(" seconds");


  /*** Restart continuous measurement with a pressure offset from 700 to 1400 millibar.
   * Giving no argument or setting the offset to 0 will disable offset correction
   */
  // if (!scd30.startContinuousMeasurement(15)){
  //   Serial.println("Failed to set ambient pressure offset");
  //   while(1){ delay(10);}
  // }
  Serial.print("Ambient pressure offset: ");
  Serial.print(scd30.getAmbientPressureOffset());
  Serial.println(" mBar");


  /*** Set an altitude offset in meters above sea level.
   * Offset value stored in non-volatile memory of SCD30.
   * Setting an altitude offset will override any pressure offset.
   */
  if (!scd30.setAltitudeOffset(360)){
    Serial.println("Failed to set altitude offset");
    while(1){ delay(10);}
  }
  Serial.print("Altitude offset: ");
  Serial.print(scd30.getAltitudeOffset());
  Serial.println(" meters");


  /*** Set a temperature offset in hundredths of a degree celcius.
   * Offset value stored in non-volatile memory of SCD30.
   */
  if (!scd30.setTemperatureOffset(175)){ // 19.84 degrees celcius
    Serial.println("Failed to set temperature offset");
    while(1){ delay(10);}
  }
  Serial.print("Temperature offset: ");
  Serial.print((float)scd30.getTemperatureOffset()/100.0);
  Serial.println(" degrees C");


  /*** Force the sensor to recalibrate with the given reference value
   * from 400-2000 ppm. Writing a recalibration reference will overwrite
   * any previous self calibration values.
   * Reference value stored in non-volatile memory of SCD30.
   */
  // if (!scd30.forceRecalibrationWithReference(400)){
  //   Serial.println("Failed to force recalibration with reference");
  //   while(1) { delay(10); }
  // }
  Serial.print("Forced Recalibration reference: ");
  Serial.print(scd30.getForcedCalibrationReference());
  Serial.println(" ppm");

Here is the example I used, I changed some values like interval, temperature offset and altitude

  /*** Enable or disable automatic self calibration (ASC).
   * Parameter stored in non-volatile memory of SCD30.
   * Enabling self calibration will override any previously set
   * forced calibration value.
   * ASC needs continuous operation with at least 1 hour
   * 400ppm CO2 concentration daily.
   */
  // if (!scd30.selfCalibrationEnabled(true)){
  //   Serial.println("Failed to enable or disable self calibration");
  //   while(1) { delay(10); }
  // }
  if (scd30.selfCalibrationEnabled()) {
    Serial.print("Self calibration enabled");
  } else {
    Serial.print("Self calibration disabled");
  }

  Serial.println("\n\n");
}

void loop() {
  if (scd30.dataReady()) {
    
    if (!scd30.read()){ 
      Serial.println("Error reading sensor data"); 
      return; 
    }

    Serial.print("Temperature: ");
    Serial.print(scd30.temperature);
    Serial.println(" degrees C");
    
    Serial.print("Relative Humidity: ");
    Serial.print(scd30.relative_humidity);
    Serial.println(" %");
    
    Serial.print("CO2: ");
    Serial.print(scd30.CO2, 3);
    Serial.println(" ppm");
    Serial.println("");
  }

  delay(100);
}
```

**Comment by robinschneider** [comment]
I also tried the Sparkfun library and I get the following results there (with an Adafruit HUZZAH32)
```
SCD30 Example
Forced recalibration factor (ppm) is 400
Measurement interval (s) is 2
Temperature offfset (C) is 1.75
Altitude offset (m) is 360
Firmware version is 0x342
Auto calibration set to false
```

The Sparkfun example just reads the data.  
All values reported by their library seem to be fine to me, these are the exact values I set before with the Adafrui library, except the measurement interval, but this one is not persistant, so it is fine.

**Comment by kevinjwalters** [comment]
This does not look relevant for this ticket but it's interesting...

I stumbled across this recently: [Sensirion SCD 30: different CO2 readings depending on measurement interval](https://www.rainer-gerhards.de/2021/01/sensirion-scd-30-different-co2-readings-depending-on-measurement-interval/)

I gave away the SCD-30s I had so no chance to compare and I've only skimmed article. Worth a thorough read and checking the code to see if it's changing the sensor's interval or just polling at different rates with a fixed interval.

**Comment by caternuson** [comment]
Oops. Got auto-closed because of comment in PR. Reopening until we verify this is fixed.

**Comment by caternuson** [comment]
@robinschneider Please try updating to the 1.0.7 release version of the library.

**Comment by robinschneider** [comment]
So I tested 1.0.7 today and it fixed the issue with the wrong interval. Now it prints the value I set. I used the sensor_tunig example and just set a altitude and temperature offset as well as a different measurement interval.

Only thing I still notice when I run the program it works, when I rebbot via button on my Huzzah32 or even when I flash the program again it says:
```
Adafruit SCD30 Sensor adjustment test!
SCD30 Found!
Failed to set measurement interval
```
When I flash the same software again or press the button again it works. This behavior happens exactly every second time.
